src/components/search/SearchField.js

```diff
@@ -0,0 +1,121 @@
+import React,  { useState, useEffect } from 'react'
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+import { Button, Modal, Card } from 'antd';
+import useModal from '../Modal/useModal';
+import SearchModalContent from '../Modal/SeachModalFragment';
+import styled from "styled-components";
+import  data from '../search-bar/MOCK_DATA.json';
+
+const SearchField = ({ value }) => {
+    //state variables
+    const [ toggle, setToggle ] = useState(false);
+    const [ options, setOptions ] = useState();
+    const [ list, setList ] = useState([]);
+    //custom hooks
+    const SearchModal = Modal;
```

Comment:
I don't understand the reassignment here. If we're worried about passing by reference than this wouldn't eliminate that unless I'm missing something here.

Reply:
Hey Will, Thanks for looking over this. So the way I'm bringing in the options are sort of in the component did mount phase , when a user first loads onto the page. I never edit those options because if someone is searching and they press back space or some other edition , then if I directly filtered down that list it wouldn't have all the true possible results , so I created a separate list that gets assigned the value of the filtered options array, and in that function I reset it's value so I don't have to worry about re-inputs or mutating the returned data. Doing it this way also helps keep from calling the API every key press the user types in , so it should perform much faster by that aspect as well. The list and options are two different states being managed - options that raw uncut version and the list is wherever the users are at search wise. If there's a better way to do this without breaking the code with a continuous use-Effect or calling the API every time I'm open to it, but from my view point this was an efficient way of cycling through the data with the most performant run time.

Reply:
This was adapted from some preexisting code, reassignment included. It doesn't appear to do anything, so we should be able to eliminate that safely.